CMakeLists.txt

```diff
@@ -49,14 +55,19 @@ target_include_directories(xiloader PUBLIC ${PROJECT_SOURCE_DIR}/xiloader)
 
 target_link_libraries(xiloader PUBLIC
     detours
-	argparse
-	crypt32
-	psapi
-	ws2_32
-	iphlpapi
-	MbedTLS::mbedtls
-	MbedTLS::mbedcrypto
-	MbedTLS::mbedx509)
+    argparse
+    crypt32
+    psapi
+    ws2_32
+    iphlpapi
+    MbedTLS::mbedtls
+    MbedTLS::mbedcrypto
+    MbedTLS::mbedx509    
+    nlohmann_json::nlohmann_json
+    PRIVATE ftxui::screen
+    PRIVATE ftxui::dom
+    PRIVATE ftxui::component
```

Comment:
We're not going on to link or expose `xiloader`, so these don't need to be marked as private. If anything this block could be: `target_link_libraries(xiloader PRIVATE`